src/Lumper.UI/Updater/Updater.cs

```diff
@@ -0,0 +1,406 @@
+
+       return new Tuple<bool, Version>(current != latest, latest);
+    }
+    public static async Task<Stream?> HttpDownload(string url, string fileName, IoProgressWindow progressWindow, IoHandler handler, CancellationTokenSource cts)
+    {
+        //the amount the progress bar should go up to while downloading - we still have to unzip for the other percentage
+        var buffer = System.Buffers.ArrayPool<byte>.Shared.Rent(80 * 1024);
+
+        if (File.Exists(fileName))
+            File.Delete(fileName);
+
+        var stream = new FileStream(fileName, FileMode.CreateNew);
+        try
+        {
+
+            using var httpClient = new HttpClient();
+            using HttpResponseMessage response =
+                await httpClient.GetAsync(url, HttpCompletionOption.ResponseHeadersRead, cts.Token);
+
+            await using Stream downloadStream = await response.Content.ReadAsStreamAsync(cts.Token);
+
+            if (response.Content.Headers.ContentLength is null)
+            {
+                handler.UpdateProgress(0, "Downloading (unknown length)");
+                await downloadStream.CopyToAsync(stream, cts.Token);
+            }
+            else
+            {
+                int read;
+                var length = (int)response.Content.Headers.ContentLength.Value;
+                var remaining = length;
+                while (!handler.Cancelled &&
+                       (read = await downloadStream.ReadAsync(
+                           buffer.AsMemory(0, int.Min(buffer.Length, remaining)),
+                           cts.Token)) >
+                       0)
+                {
+                    var prog = (float)read / length * DownloadProgressPercentage;
+                    handler.UpdateProgress(prog, $"{float.Floor((1 - ((float)remaining / length)) * DownloadProgressPercentage)}%");
+                    await stream.WriteAsync(buffer.AsMemory(0, read), cts.Token);
+                    remaining -= read;
+                }
+            }
+        }
+        catch (Exception ex)
+        {
+            if (ex is TaskCanceledException)
+                Logger.Info("Download cancelled by user");
+            else
+                Logger.Error(ex, "Failed to download!");
+
+            await stream.DisposeAsync();
+            return null;
+        }
+        finally
+        {
+            stream.Close();
+        }
+
+        if (handler.Cancelled)
+            return null;
+
+        return stream;
+    }
+    /// <summary>
+    /// Returns the URL to the download link for the OS-specific version.
+    /// </summary>
+    /// <returns></returns>
+    private static string GetPath(GithubUpdate assets, OSPlatform OS)
+    {
+        var operatingSystem = "";
+
+        if (OS == OSPlatform.Windows)
+            operatingSystem = "win";
+        else if (OS == OSPlatform.Linux)
+            operatingSystem = "linux";
+        else
+            throw new ApplicationException("Unsupported OS: " + OS);
```

Comment:
We're picking this same string in `Update`, so let's just pass that in to this function. I'll post a rewritten version of `Update` below, you can modify this function to take a string input.